### ISSUE TITLE
Attribute Count Wrong in layered navigation

### ISSUE THREAD

**Issue by petersjeroen** [woocommerce/woocommerce#17465]
When using the "Filter products by attribute" in the layered navigation, the count is wrong.
I made a screenshot to clarify the problem: 

![screenshot](https://user-images.githubusercontent.com/33237012/32215220-a87ddbd0-be21-11e7-8524-6f8086f823ce.jpg)

## Prerequisites
- [x] I have searched for similar issues in both open and closed tickets and found this similar issue: https://github.com/woocommerce/woocommerce/issues/6583

## What I have tried myself

- [x] Removed the attribute from each product and added it again
- [x] Pressed the "Recount terms" button - WooCommerce --> Status --> Tools --> Term counts
- [x] Removed the "Filter Products by Attribute" widget and added it again
- [x] Added a new term to the "Color" attribute and assigned this to a product. Please take a look at the YouTube video on my next comment in this issue.

## Steps to reproduce the issue

1. Install Wordpress + WooCommerce (fresh installation, without any plugins)
2. Import sample data
3. Provide some products with the attribute "Black"
4. Activate the "Filter Products by Attribute" widget in the sidebar. Title: Color, Attribute: color, Display type: List, Query Type: AND - Save

## Isolating the problem
- [x] This bug happens with the default  [Storefront] theme (https://woocommerce.com/storefront/)
- [x] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
```
`
### WordPress Environment ###

Home URL: http://localhost:8888/testwp
Site URL: http://localhost:8888/testwp
WC Version: 3.2.1
Log Directory Writable: ✔
WP Version: 4.8.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 7.1.1
PHP Post Max Size: 32 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.52.1
OpenSSL/1.0.2j

SUHOSIN Installed: –
MySQL Version: 5.6.35
Max Upload Size: 32 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.2.1
WC Database Prefix: wp_
MaxMind GeoIP Database: ❌ The MaxMind GeoIP Database does not exist - Geolocation will not function. You can download and install it manually from http://dev.maxmind.com/geoip/legacy/geolite/ to the path: . Scroll down to "Downloads" and download the "Binary / gzip" file next to "GeoLite Country". Please remember to uncompress GeoIP.dat.gz and upload the GeoIP.dat file only.
Total Database Size: 1.90MB
Database Data Size: 0.98MB
Database Index Size: 0.92MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.05MB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB
wp_comments: Data: 0.02MB + Index: 0.08MB
wp_links: Data: 0.02MB + Index: 0.02MB
wp_options: Data: 0.22MB + Index: 0.02MB
wp_postmeta: Data: 0.22MB + Index: 0.16MB
wp_posts: Data: 0.08MB + Index: 0.06MB
wp_termmeta: Data: 0.02MB + Index: 0.03MB
wp_terms: Data: 0.02MB + Index: 0.03MB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB
wp_usermeta: Data: 0.02MB + Index: 0.03MB
wp_users: Data: 0.02MB + Index: 0.05MB

### Post Type Counts ###

attachment: 68
customize_changeset: 1
nav_menu_item: 7
page: 7
post: 2
product: 36
product_variation: 4

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 3.2.1

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: EUR (€)
Currency Position: left
Thousand Separator: 
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #4 - /winkel/
Cart: #5 - /winkelmand/
Checkout: #6 - /afrekenen/
My account: #7 - /mijn-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.2.5
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –
`
```
</details>

**Comment by petersjeroen** [comment]
When adding a new term to the "Color" attribute, the count is displaying correctly. As soon as I remove that new term, the count is showing wrong again. Please take a look at this YouTube video where I demonstrate this: https://www.youtube.com/watch?v=3dJrr9ODyus

**Comment by mikejolley** [comment]
This is a caching issue being fixed in 3.2.2 today.
